docs/contributing/unstructured-source.md

```diff
@@ -0,0 +1,614 @@
+# Unstructured Source
+
+Unstructured source provides a generic mechanism to manage DNS records in your favourite DNS provider supported by external-dns by watching for a user specified Kubernetes API to extract hostnames, targets, TTL, and record type information from a combination of well-known annotations and evaluating JSONPath expressions against the resource.
+
+* [Details](#details)
+* [Usage](#usage)
+* [RBAC Configuration](#rbac-configuration)
+* [Demo](#demo)
+  * [Setup](#setup-steps)
+  * [Examples](#examples):
+    * `ConfigMap`
+      * [DNS A-record with multiple targets](#dns-a-record-with-multiple-targets-from-a-configmap)
+      * [Multiple DNS A-records](#multiple-dns-a-records-from-a-configmap)
+    * `Deployment`
+      * [DNS CNAME-record from annotations](#dns-cname-record-from-annotation-on-a-deployment)
+    * `Workload` (_namespace-scoped CRD_)
+      * [DNS A-record](#dns-a-record-from-a-namespace-scoped-crd)
+      * [Multiple DNS A-records](#multiple-dns-a-records-from-a-namespace-scoped-crd)
+      * [DNS A-record with multiple targets](#dns-a-record-with-multiple-targets-from-a-namespace-scoped-crd)
+    * `ClusterWorkload` (_cluster-scoped CRD_)
+      * [DNS A-record](#dns-a-record-a-cluster-scoped-crd)
+  * [Cleanup](#cleanup-steps)
+
+## Details
```

Comment:
Because it seems very different than the envisioned purpose of external-dns, can you please add more about architecture, when it makes sense to use when it does not?
There is a need for more context here. I basically don't understand who would use it and why one would use this instead of the current available sources. What do you do better or enable (use cases?), than the current offer?